File: Leetcode/math/1175_PrimeArrangements/Solution.java
import java.math.BigInteger;
class Solution {
    public int numPrimeArrangements(int n) {
        if(n<=2)
            return 1;
        int primeCount = 0;
        for(int i = 2; i<=n; i++){
            if(isPrime(i))
                primeCount++;
        }
        BigInteger a = factorial(primeCount);
        BigInteger b = factorial(n-primeCount);
        BigInteger res = a.multiply(b);
        return res.mod(BigInteger.valueOf(1_000_000_007)).intValue();
    }
        private BigInteger factorial(int n){
            BigInteger fact = BigInteger.valueOf(1);
            for (int i = 1; i <= n; i++) {
               fact = fact.multiply(BigInteger.valueOf(i));
            }
            return fact;
        }
        private boolean isPrime(int n) {
            int count = 0;
            for(int i = 1; i <= n; i++) {
                if(n % i == 0) {
                    count++;
                }
                if(count > 2) {
                    return false;
                }
            }
            return true;
        }
}

File: Leetcode/math/1175_PrimeArrangements/Solution.py
class Solution(object):
    def numPrimeArrangements(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n<=2:
            return 1
        primeCount = self.countPrimes(n+1)
        a = self.fac(primeCount)
        b = self.fac(n-primeCount)
        return a*b%1000000007

    def fac(self, n):
        return 1 if n<=1 else n*self.fac(n-1)

    def countPrimes(self, n):
        if n < 2:
            return 0
        s = [1] * n
        s[0] = s[1] = 0
        for i in range(2, int(n ** 0.5) + 1):
            if s[i] == 1:
                s[i * i:n:i] = [0] * len(s[i * i:n:i])
        return sum(s)
